### ISSUE TITLE
Passing job parameters into Docker build arguments does not work

### ISSUE THREAD

**Issue by Aleksei-Badyaev** [jenkinsci/docker-plugin#903]
### Jenkins and plugins versions report

<details>
  <summary>Environment</summary>

  
  ```text
  Paste the output here
  ```

</details>

### What Operating System are you using (both controller, and any agents involved in the problem)?

Debian 11, jenkins run in official docker image `jenkins/jenkins`.

### Reproduction steps

1. Create parametrized job and set up some parameter, NAME=VALUE for example.
2. Configure Docker build/publish stage in build with build args with content: `--build-arg NAME=${NAME}
3. Run job

### Expected Results

Build args passed to docker build command in form `--build-arg NAME=VALUE`

### Actual Results

Build args passed to docker build command in form `--build-arg NAME=${NAME}`, which no sense.

### Anything else?

With this bug, the Docker plugin is almost useless for building images parameterized with job arguments.

**Comment by rimvislt** [comment]
"--build-arg" passed through agent from credentials does not seem to be working either, so what is solution with build arguments to pass credentials?

**Comment by Peter-Darton-i2** [comment]
TL;DR: "not this"

You could use this plugin to spin up a container from which you can run `docker build` commands from a simple shell step.
i.e. use this plugin purely as a means of providing Jenkins agent executors; that's what it's designed to do.

FYI the "build step" functionality this plugin contains is not really "core functionality" to this plugin and maintaining/enhancing these various build steps has never been a priority; there's a strong argument that these build steps should be split out into a separate plugin ... but nobody's volunteered to do that work either.
